### ISSUE TITLE
[flatpak-pip-generator] 'get_tar_package_url_pypi' error with multiple packages.

### ISSUE THREAD

**Issue by otrocodigo** [flatpak/flatpak-builder-tools#416]
### flatpak-builder version

1.2.3

### Linux distribution and version

Linux debian 6.1.0-22-amd64

### Affected flatpak-builder tool

pip/flatpak-pip-generator

### Description

When trying to generate the 'mediapipe' manifest source I detected errors with several Python packages:

- jaxlib-0.4.30
`python flatpak-pip-generator jaxlib`
```
Traceback (most recent call last):
  File "$PATH/flatpak-pip-generator", line 291, in <module>
    url = get_tar_package_url_pypi(name, version)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PATH/flatpak-pip-generator", line 91, in get_tar_package_url_pypi
    raise Exception(err)
Exception: Failed to get jaxlib-0.4.30 source from https://pypi.org/pypi/jaxlib/0.4.30/json
```

- torch-2.3.1

```
Traceback (most recent call last):
  File "$PATH/flatpak-pip-generator", line 291, in <module>
    url = get_tar_package_url_pypi(name, version)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PATH/flatpak-pip-generator", line 91, in get_tar_package_url_pypi
    raise Exception(err)
Exception: Failed to get torch-2.3.1 source from https://pypi.org/pypi/torch/2.3.1/json
```
- mediapipe-0.10.13
`python flatpak-pip-generator mediapipe==0.10.13`
```
Traceback (most recent call last):
  File "$PATH/flatpak-pip-generator", line 291, in <module>
    url = get_tar_package_url_pypi(name, version)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "$PATH/flatpak-pip-generator", line 91, in get_tar_package_url_pypi
    raise Exception(err)
Exception: Failed to get mediapipe-0.10.13 source from https://pypi.org/pypi/mediapipe/0.10.13/json
```
---

Basically, I detect errors with 'mediapipe' dependencies, I tried several versions and some dependency always fails.

**Comment by hfiguiere** [comment]
you should also complain to the package for not providing the source packages.

**Comment by hfiguiere** [comment]
granted the error should be better handled to actually say what it is.

**Comment by otrocodigo** [comment]
@sgvictorino and @hfiguiere Thank you. I didn't know about that limitation.
